modules/padsquadBidAdapter.js

```diff
@@ -0,0 +1,133 @@
+        exchange: {
+          unitId: bidRequest.params.unitId
+        }
+      }
+    }));
+
+    const openrtbRequest = {
+      id: bidderRequest.auctionId,
+      imp: impressions,
+      site: {
+        domain: window.location.hostname,
+        page: window.location.href,
+        ref: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer || null : null
+      },
+      ext: {
+        exchange: {
+          publisherId: publisherId,
+          networkId: networkId,
+        }
+      }
+    };
+
+    // apply gdpr
+    if (bidderRequest.gdprConsent) {
+      openrtbRequest.regs = {ext: {gdpr: bidderRequest.gdprConsent.gdprApplies ? 1 : 0}};
+      openrtbRequest.user = {ext: {consent: bidderRequest.gdprConsent.consentString}};
+    }
+
+    const payloadString = JSON.stringify(openrtbRequest);
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL,
+      data: payloadString,
+    };
+  },
+
+  interpretResponse: function (serverResponse, request) {
+    const bidResponses = [];
+    const response = (serverResponse || {}).body;
+    // response is always one seat with (optional) bids for each impression
+    if (response && response.seatbid && response.seatbid.length === 1 && response.seatbid[0].bid && response.seatbid[0].bid.length) {
+      response.seatbid[0].bid.forEach(bid => {
+        bidResponses.push({
+          requestId: bid.impid,
+          cpm: bid.price,
+          width: bid.w,
+          height: bid.h,
+          ad: bid.adm,
+          ttl: DEFAULT_BID_TTL,
+          creativeId: bid.crid,
+          netRevenue: DEFAULT_NET_REVENUE,
+          currency: DEFAULT_CURRENCY,
+        })
+      })
+    } else {
+      utils.logInfo('padsquad.interpretResponse :: no valid responses to interpret');
+    }
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions, serverResponses) {
+    utils.logInfo('padsquad.getUserSyncs', 'syncOptions', syncOptions, 'serverResponses', serverResponses);
+    let syncs = [];
+
+    if (!syncOptions.iframeEnabled && !syncOptions.pixelEnabled) {
+      return syncs;
+    }
+
+    serverResponses.forEach(resp => {
+      const userSync = utils.deepAccess(resp, 'body.ext.usersync');
+      if (userSync) {
+        let syncDetails = [];
+        Object.keys(userSync).forEach(key => {
+          const value = userSync[key];
+          if (value.syncs && value.syncs.length) {
+            syncDetails = syncDetails.concat(value.syncs);
+          }
+        });
+        syncDetails.forEach(syncDetails => {
+          syncs.push({
+            type: syncDetails.type === 'iframe' ? 'iframe' : 'image',
+            url: syncDetails.url
+          });
+        });
+
+        if (!syncOptions.iframeEnabled) {
+          syncs = syncs.filter(s => s.type !== 'iframe')
+        }
+        if (!syncOptions.pixelEnabled) {
+          syncs = syncs.filter(s => s.type !== 'image')
+        }
+      }
+    });
+    utils.logInfo('padsquad.getUserSyncs result=%o', syncs);
```

Comment:
The log message is a bit ambiguous. Would you mind fixing that or making it more coherent as to what it means?

Reply:
actually that line may not even be necessary, we'll remove it altogether